### PR TITLE
Pmyip3/be able to create/edit/delete publications via frontend

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Create UI based to the wireframe to allow create, edit and delete publication
Note: also combined publications and tweets on the publication page 

# Type of change 
- [ ] Bug fix
- [ X ] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
Currently the publication page only display a basic page of publicaitons but there's no way for users to create, edit and edit publicaiton via the frontend 

# Solution description
Code changes in this PR improve the UI of the publication page, also allowing users to create, edit and delete publicaiton.
note: import publication does not connection to backend at the moment 

# Tests 
All unit tests pass? No unit tests implemented at the moment 
Build successfully? Yes  <br/> 
Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes) <br/>
Publicaitons page
image

Add publicaiton button clicked
image

Import publication button clicked
image

Expand arrow icon clicked 
image

Vertical three dots icon clicked
image

Edit pulibcaiton button clicked
image

Delete publicaiton button clicked
image

Form validation 
image
image

Creating, editing and deleting also result in a change in database

** update
Combining publications and tweet on the publication page 
image

# Improvent for later
- consider the case when no publication ( show a message )
- animation when the expand arrow is clicked
- displaying a spinning icon when fetching

# Relevant document/ link (if any) 
Wireframes doc: https://drive.google.com/drive/u/2/folders/1Vjj3Lvodg3L3wKc9nPO9549iP28UO-y0

# Risk
high risk, publicaiton cannot be retrieved, created, edited or deleted if bug occurs
relevant components could be affected: Publication.js, Publications.js, PublicationForm.js

### CODE REVIEW

**Comment by rsun0013** [comment]
It says there are conflicts, could you merge from Scholar into your branch and resolve them?

**Review by vgoh1 — COMMENTED** [review]
2 things:

Add a url validator for this:
image
Since it's fixed to https right now, I think it'd be better to give the user an option to paste in a link and then validate that cause what if its http

The form validation behaves weird:
image
The validation happens only if you backspace from ^, this needs to be fixed

**Review by aksh0001 — COMMENTED** [review]
I was thinking, should we merge your Publications UI in the placeholder container I created when rendering tweets (in this PR: https://github.com/Researchify/Researchify/pull/69)?

**Comment by pmyip3** [comment]
> I was thinking, should we merge your Publications UI in the placeholder container I created when rendering tweets (in this PR: #69)?

Sure. I can merge it after your ticket merged to our team branch. 
May be we need to rename the TwitterPanel.js file as well if we decided rendering 2 components in this file.

**Comment by aksh0001** [comment]
> > I was thinking, should we merge your Publications UI in the placeholder container I created when rendering tweets (in this PR: #69)?
> 
> Sure. I can merge it after your ticket merged to our team branch.
> May be we need to rename the TwitterPanel.js file as well if we decided rendering 2 components in this file.

yep, that is now merged. And yeah feel free to create the container (or rename components) in the publications component as well or as best you see fit.

**Comment by pmyip3** [comment]
> 2 things:
> 
> Add a url validator for this:
> image
> Since it's fixed to https right now, I think it'd be better to give the user an option to paste in a link and then validate that cause what if its http
> 
> The form validation behaves weird:
> image
> The validation happens only if you backspace from ^, this needs to be fixed

Should be all good now

File: client/src/api/api.js
Comment (aksh0001) [inline client/src/api/api.js:10]: curious: can we leave this to `true`? I made this PR that sets up a proxy server for development: https://github.com/Researchify/Researchify/pull/65. With this, we can set this to true always. Let me know if it doesn't work for you

**Review by aksh0001 — APPROVED** [review]
Maybe look into the the usingRp var. But otherwise, nice!

File: client/src/api/api.js
Comment (pmyip3) [reply client/src/api/api.js:10]: Sure. Will get it fixed

**Pull request merged**
